src/web/routes/application/select-address/adapters.js

```diff
@@ -0,0 +1,75 @@
+const { props, propOr, pickAll, compose, map, join, filter, not, isEmpty, split, head, tail, defaultTo } = require('ramda')
+const { THOROUGHFARE_PROPS, BUILDING_NAME_PROPS, LOCALITY_PROPS, DELIMITER } = require('./constants')
+const { OS_PLACES_ADDRESS_KEYS } = require('../common/constants')
+
+/**
+ * Map address returned from OS places API to format required by claimant service
+ * N.B. assumes address has been normalised by /postcode adapter
+ *
+ * 1. Normalise address by setting any missing fields to a default value of an empty string
+ * 2. Construct the 4 possible address parts for address lines 1 and 2:
+ *    a) Organisation name
+ *    b) Building name
+ *    c) Number and street
+ *    d) Locality
+ * 3. Split a single part address into two parts if a comma exists in the first part
+ * 4. Convert a multipart address into two parts by joining all parts except the first with a comma
+ */
+
+const isNotEmpty = compose(not, isEmpty)
+
+const joinAddressFields = addressKeys => compose(join(DELIMITER), filter(isNotEmpty), props(addressKeys))
+
+const constructThoroughfare = joinAddressFields(THOROUGHFARE_PROPS)
```

Comment:
I found this (and the two below) constant quite confusing, if its not used anywhere else, could we inline them here, I think seeing `joinAddressFields['DEPENDENT_THOROUGHFARE_NAME', 'THOROUGHFARE_NAME']` here would be much clearer.

Reply:
Sure - I thought it was cleaner to extract these to constants (as the actual prop names seemed irrelevant) but happy to inline

Reply:
I've moved these constants inline